ui/js/controllers/bugfiler.js

```diff
@@ -0,0 +1,245 @@
+            for(var i=0;i<allFailures.length;i++) {
+                var omittedLeads = ["TEST-UNEXPECTED-FAIL", "PROCESS-CRASH", "TEST-UNEXPECTED-ERROR", "TEST-UNEXPECTED-TIMEOUT"];
+                for(var j=0; j < omittedLeads.length; j++) {
+                    if(allFailures[i][0].search(omittedLeads[j]) >= 0) {
+                        allFailures[i].shift();
+                    }
+                }
+                if(i != 0) {
+                    thisFailure += "\n";
+                }
+                thisFailure += allFailures[i].join(" | ");
+                $("#modalFailureList");
+            }
+            $("#modalFailureList").val(thisFailure);
+
+            $scope.findProduct();
+        };
+
+        $modalInstance.parsedSummary = "";
+        $modalInstance.initiate = $scope.initiate;
+        $modalInstance.possibleFilename = "";
+
+        /*
+         *  Remove extraneous junk from the start of the summary line
+         *  and try to find the failing test name from what's left
+         */
+        $modalInstance.parseSummary = function(summary) {
+            var omittedLeads = ["TEST-UNEXPECTED-FAIL", "PROCESS-CRASH", "TEST-UNEXPECTED-ERROR", "TEST-UNEXPECTED-TIMEOUT"];
+            summary = summary.split(" | ");
+
+            for(var i=0; i < omittedLeads.length; i++) {
+                if(summary[0].search(omittedLeads[i]) >= 0) {
+                    summary.shift();
+                }
+            }
+    // XXX Maybe get crash signatures too?
+            $modalInstance.possibleFilename = summary[0].split("/").pop();
+
+            return [summary, $modalInstance.possibleFilename];
+        };
+
+        /*
+         *  Toggle the visibility of the rest of the lines from the failure summary
+         */
+        $scope.toggleFailures = function(evt) {
+            var target = evt.target;
+            if(target.tagName.toUpperCase() == "I") {
+                target = target.parentNode;
+            }
+            if(target.className == "btn btn-xs failure-expando-closed") {
+                target.className = "btn btn-xs failure-expando-open";
+                target.firstElementChild.className = "fa fa-chevron-down";
+                $("#failureSummaryGroup")[0].className = "expanded";
+            } else {
+                target.className = "btn btn-xs failure-expando-closed";
+                target.firstElementChild.className = "fa fa-chevron-right";
+                $("#failureSummaryGroup")[0].className = "collapsed";
+            }
+        };
+
+        /*
+         *  Attempt to find a good product/component for this failure
+         */
+        $scope.findProduct = function() {
+
+            var suggestedProducts = [];
+            var failurePath = $modalInstance.parsedSummary[0][0];
+            var failurePathRoot = failurePath.split("/")[0];
+
+            // Look up the product via the root of the failure's file path
+            // XXX THIS NEEDS TO BE MUCH MORE ROBUST
```

Comment:
By Athena's Toothpaste, could this be a little more robust, please?  :) Or if it's robust enough, should we remove the comment?